manifests/install.pp

```diff
@@ -14,9 +14,6 @@
       ensure => "directory";
   }
 
-  systemd::unit_file {"${nats::service_name}.service":
-    content => epp("nats/systemd_service.epp")
-  }
-
+  Class["nats::install::${nats::service_type}"] ~> Class["nats::service"]
```

Comment:
might be easier to have `nat::install` which inside there contains the `nats::install::${nats::service_type}` this way you always just require `Class["nats::service"]` and avoid some of these extra chaining?

If it seems about the same or worse no biggie just thought it might be a bit better